onadata/libs/serializers/share_project_serializer.py

```diff
@@ -20,14 +20,28 @@ def attrs_to_instance(attrs, instance):
 
 class ShareProjectSerializer(serializers.Serializer):
     project = ProjectField()
-    username = serializers.CharField(max_length=255)
+    username = serializers.CharField(max_length=255,
+                                     allow_null=True,
+                                     required=False)
+    usernames = serializers.CharField(required=False)
```

Comment:
Could we simply have overloaded the username field? Where the username can have a single username or a list of comma-separated usernames? This looks odd for me, I think there is no need for a new variable that has a plural meaning.

Reply:
The `usernames` field was mostly added there so it doesn't confuse new developers / users of the api but I also think overloading the username field would remove a bit of the complexity that adding the `usernames` field brings. I can make that change and document that the `username` field accepts CSV data too

Reply:
`username` with comma separated values is a bit odd to me. Personally, I'd rather have separate fields.

Reply:
After discussion it was that the `username` field shall be overloaded